### PR TITLE
[QA] remove some magic mining constants in functional tests

### PR DESCRIPTION
**Pull request by instagibbs**
The fewer magic numbers the better.

Also more directly tested a `submitheader` case of bad previous blockhash.

### CODE REVIEW

File: test/functional/mining_basic.py [hunk 1]
```diff
@@ -164,9 +164,9 @@ def assert_submitblock(block, result_str_1, result_str_2=None):
         assert_submitblock(bad_block, 'prev-blk-not-found', 'prev-blk-not-found')
 
         self.log.info('submitheader tests')
-        assert_raises_rpc_error(-22, 'Block header decode failed', lambda: node.submitheader(hexdata='xx' * 80))
-        assert_raises_rpc_error(-22, 'Block header decode failed', lambda: node.submitheader(hexdata='ff' * 78))
-        assert_raises_rpc_error(-25, 'Must submit previous header', lambda: node.submitheader(hexdata='ff' * 80))
+        assert_raises_rpc_error(-22, 'Block header decode failed', lambda: node.submitheader(hexdata='xx' * BLOCK_HEADER_SIZE))
+        assert_raises_rpc_error(-22, 'Block header decode failed', lambda: node.submitheader(hexdata='ff' * (BLOCK_HEADER_SIZE-2)))
+        assert_raises_rpc_error(-25, 'Must submit previous header', lambda: node.submitheader(hexdata=super(CBlock, bad_block).serialize().hex()))
```
Comment (practicalswift) [inline test/functional/mining_basic.py]: I'm not sure I follow here: which constructor is it that takes `CBlock` and `bad_block` as parameters?

File: test/functional/interface_rest.py [hunk 2]
```diff
@@ -211,28 +213,30 @@ def run_test(self):
         self.test_rest_request('/block/{}'.format(bb_hash))
         self.nodes[0].reconsiderblock(bb_hash)
 
+        BLOCK_HEADER_SIZE = len(CBlockHeader().serialize())
```
Comment (laanwj) [inline test/functional/interface_rest.py]: As this constant is used in multiple places, wouldn't it be an idea to define it in a shared place?

File: test/functional/mining_basic.py (hunk 1 above)
Comment (instagibbs) [reply test/functional/mining_basic.py]: The base class is CBlockHeader, so we're accessing its serialization function.

File: test/functional/interface_rest.py (hunk 2 above)
Comment (instagibbs) [reply test/functional/interface_rest.py]: (and added an assert right after for the invariant of 80)

**Comment by MarcoFalke** [comment]
utACK b651ef7e1c39a820089695b29d14a07d910a385a

File: test/functional/mining_basic.py (hunk 1 above)
Comment (Empact) [reply test/functional/mining_basic.py]: Maybe rename `bad_block` above to `bad_hashPrevBlock_block` or similar, and invoke the `CBlockHeader` constructor directly?

**Comment by Empact** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/15238/commits/b651ef7e1c39a820089695b29d14a07d910a385a

File: test/functional/mining_basic.py [hunk 3]
```diff
@@ -15,6 +15,7 @@
 from test_framework.messages import (
     CBlock,
     CBlockHeader,
+    BLOCK_HEADER_SIZE
```
Comment (MarcoFalke) [inline test/functional/mining_basic.py]: style-nit: Missing trailing `,` :eyes:

File: test/functional/mining_basic.py (hunk 1 above)
Comment (AkioNak) [inline test/functional/mining_basic.py]: Currentry, minimum required version of Python (test) is 3.4.
https://github.com/bitcoin/bitcoin/blob/master/doc/dependencies.md

But hex() seems to be introdued in Python 3.5.
https://docs.python.org/3.5/library/stdtypes.html?highlight=bytes%20hex#bytes.hex

I got following error when executing this test on Python3.4.
```
AttributeError: 'bytes' object has no attribute 'hex'
```

**Comment by Sjors** [comment]
This test also fails for me locally on Python 3.4. If you install [PyEnv](https://github.com/pyenv/pyenv) it will catch this.

**Pull request merged**